packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -0,0 +1,641 @@
+						)
+						.join("")}`,
+					direction: "inbound",
+				});
+			}
+		} else {
+			supportedSubsystems =
+				node.getValue<SubsystemType[]>(
+					getSupportedSubsystemTypesValueId(this.endpointIndex),
+				) ?? [];
+		}
+
+		for (const subsystemType of supportedSubsystems) {
+			this.driver.controllerLog.logNode(node.id, {
+				message: `querying event signaling state for subsystem ${getEnumMemberName(
+					SubsystemType,
+					subsystemType,
+				)}...`,
+				direction: "outbound",
+			});
+			const state = await api.getEventSignaling(subsystemType);
+			if (state != undefined) {
+				this.driver.controllerLog.logNode(node.id, {
+					message: `subsystem ${getEnumMemberName(
+						SubsystemType,
+						subsystemType,
+					)} has state ${getEnumMemberName(SubsystemState, state)}`,
+					direction: "inbound",
+				});
+			}
+		}
+
+		this.driver.controllerLog.logNode(node.id, {
+			message: "querying current barrier state...",
+			direction: "outbound",
+		});
+		await api.get();
+
+		// Remember that the interview is complete
+		this.interviewComplete = true;
+	}
+}
+
+interface BarrierOperatorCCSetOptions extends CCCommandOptions {
+	state: BarrierState.Open | BarrierState.Closed;
+}
+
+@CCCommand(BarrierOperatorCommand.Set)
+export class BarrierOperatorCCSet extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| BarrierOperatorCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.state = options.state;
+		}
+	}
+
+	public state: BarrierState.Open | BarrierState.Closed;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([this.state]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: { "target state": this.state },
+		};
+	}
+}
+
+@CCCommand(BarrierOperatorCommand.Report)
+export class BarrierOperatorCCReport extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+
+		// return values state and position value
+		// if state is 0 - 99 or FF (100%) return the appropriate values.
+		// if state is different just use the table and
+		// return undefined position
+
+		const payloadValue = this.payload[0];
+		this.state = payloadValue;
+		this.position = undefined;
+		if (payloadValue <= 99) {
+			this.position = payloadValue;
+			if (payloadValue > 0) {
+				this.state = undefined;
+			}
+		} else if (payloadValue === 255) {
+			this.position = 100;
+			this.state = payloadValue;
+		}
+
+		this.persistValues();
+	}
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
```

Comment:
Not familiar with typescript or this code much yet so... does this ReadOnlyUInt8 mean the value id is not writeable? If so, is this value supposed to be used for reporting and setting the state?

Reply:
good catch. That implies `writeable: false`.